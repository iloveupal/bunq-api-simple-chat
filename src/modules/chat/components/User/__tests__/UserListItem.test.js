import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';

import UserListItem from '../UserListItem';


describe('UserListItem', () => {
    it('should match snapshot', () => {
        const wrapper = mount(
            <UserListItem id={'1'} name={'Test'} onClick={jest.fn()}/>
        );

        expect(wrapper).toMatchSnapshot();
    });
});