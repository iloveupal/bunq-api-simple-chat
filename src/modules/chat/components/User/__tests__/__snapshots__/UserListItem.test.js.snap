// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserListItem should match snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 40px;
  height: 40px;
  border-radius: 20px;
  background-color: #eeeeee;
  font-weight: bold;
  font-family: sans-serif;
  font-size: 14px;
  color: white;
  cursor: pointer;
}

.c3 {
  color: white;
  font-family: sans-serif;
  font-weight: bold;
  font-size: 14px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
  width: 300px;
  cursor: pointer;
}

.c0:hover {
  background-color: #333;
}

.c2 {
  padding-left: 20px;
}

<UserListItem
  id="1"
  name="Test"
  onClick={[MockFunction]}
>
  <UserListItem__Container
    onClick={[Function]}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "UserListItem__Container-sc-1ir1dzz-0",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "display:flex;flex-flow:row nowrap;align-items:center;padding:10px;width:300px;cursor:pointer;&:hover{background-color:#333;}",
            ],
          },
          "displayName": "UserListItem__Container",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "UserListItem__Container-sc-1ir1dzz-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
    >
      <div
        className="c0"
        onClick={[Function]}
      >
        <UserColorCircle
          id="1"
          name="Test"
        >
          <UserColorCircle__ColorCircle
            color="#eeeeee"
            small={false}
            title="Test"
          >
            <StyledComponent
              color="#eeeeee"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "UserColorCircle__ColorCircle-sc-1fnyyl8-0",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "display:flex;flex-flow:row nowrap;justify-content:center;align-items:center;width:",
                      [Function],
                      ";height:",
                      [Function],
                      ";border-radius:",
                      [Function],
                      ";background-color:",
                      [Function],
                      ";font-weight:bold;font-family:sans-serif;font-size:",
                      [Function],
                      ";color:white;cursor:pointer;",
                    ],
                  },
                  "displayName": "UserColorCircle__ColorCircle",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "UserColorCircle__ColorCircle-sc-1fnyyl8-0",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              small={false}
              title="Test"
            >
              <div
                className="c1"
                color="#eeeeee"
                title="Test"
              >
                T
              </div>
            </StyledComponent>
          </UserColorCircle__ColorCircle>
        </UserColorCircle>
        <UserListItem__UserNameContainer>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "UserListItem__UserNameContainer-sc-1ir1dzz-1",
                  "isStatic": true,
                  "lastClassName": "c2",
                  "rules": Array [
                    "padding-left:20px;",
                  ],
                },
                "displayName": "UserListItem__UserNameContainer",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "UserListItem__UserNameContainer-sc-1ir1dzz-1",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="c2"
            >
              <UserName
                name="Test"
              >
                <UserName__NameLabel>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "UserName__NameLabel-sc-1yge4xz-0",
                          "isStatic": true,
                          "lastClassName": "c3",
                          "rules": Array [
                            "color:white;font-family:sans-serif;font-weight:bold;font-size:14px;",
                          ],
                        },
                        "displayName": "UserName__NameLabel",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "UserName__NameLabel-sc-1yge4xz-0",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <div
                      className="c3"
                    >
                      Test
                    </div>
                  </StyledComponent>
                </UserName__NameLabel>
              </UserName>
            </div>
          </StyledComponent>
        </UserListItem__UserNameContainer>
      </div>
    </StyledComponent>
  </UserListItem__Container>
</UserListItem>
`;
